src/lightgbm/src/main/scala/LightGBMUtils.scala

```diff
@@ -339,11 +339,14 @@ object LightGBMUtils {
     var data: Option[(SWIGTYPE_p_void, SWIGTYPE_p_double)] = None
     try {
       data = Some(generateData(numRows, rowsAsDoubleArray))
+      val reference =
```

Comment:
use .orNull on the option

Reply:
ah, that's very cool!  great suggestion.  done.